ReflectometryServer/pv_wrapper.py

```diff
@@ -202,60 +200,114 @@ def sp(self, value):
 
     @property
     def backlash_distance(self):
         """
         Returns(float): the value of the underlying backlash distance PV
         """
-        if self._dir == "Pos":
-            return self._d_back * -1.0
+        if self._moving_direction == "Pos":
+            return self._backlash_distance * -1.0
         else:
-            return self._d_back
+            return self._backlash_distance
 
     @property
     def backlash_velocity(self):
         """
-        Returns: the value of the underlying backlash velocity PV
+        Returns (float): the value of the underlying backlash velocity PV
         """
-        return self._v_back
+        return self._backlash_velocity
 
     @property
     def direction(self):
         """
         Returns: the value of the underlying direction PV
         """
-        return self._dir
-
-    @velocity.setter
-    def velocity(self, value):
-        """
-        Writes a value to the underlying velocity PV's VAL field.
-
-        Args:
-            value: The value to set
-        """
-        self._write_pv(self._velo_pv, value)
-
-    def initiate_move_with_change_of_velocity(self):
-        """
-        Sets internal state of the pv wrapper to reflect that a move has just been initialised.
-        """
-        self._move_initiated = True
-        self._velocity_event.clear()
-        if self._moving_state == MTR_STOPPED:
-            self._v_restore = self.velocity
-            write_autosave_value(self.name, self._v_restore, AutosaveType.VELOCITY)
+        return self._moving_direction
+
+    def _init_velocity_cache(self):
+        """
+        Initialise the velocity cache for the current axis.
+
+        Initialise the velocity cache and its restored status. If no cache exists then load the last velocity value
+        from the auto-save file.
+        """
+        if self._velocity_cache is None:
+            try:
+                autosave_value = read_autosave_value(self.name, AutosaveType.VELOCITY)
+                if autosave_value is not None:
+                    logger.debug("Restoring {pv_name} velocity_cache with auto-save value {value}"
+                                 .format(pv_name=self.name, value=autosave_value))
+                    self._velocity_cache = float(autosave_value)
+                autosave_value = read_autosave_value(self.name+"_velocity_cache_restored", AutosaveType.VELOCITY)
+                if autosave_value is not None:
+                    logger.debug("Restoring {pv_name} velocity_cache_restored with auto-save value {value}"
+                                 .format(pv_name=self.name, value=autosave_value))
+                    self._velocity_cache_restored = bool(autosave_value)
+            except (ValueError, TypeError) as error:
+                logger.error("Error: Unable to initialise velocity cache from auto-save ({error_message})."
+                             .format(error_message=error))
+
+    def cache_velocity(self):
```

Comment:
For `JawsCentrePVWrapper` (e.g. S3 on CRISP) which does not cache velocities, this currently always prints an error on move. I think eventually we should refactor the entire PVWrapper class structure to make the distinction between synchronized / non-synchronized axes- but for now could you override this method in the subclass to just `pass` ?

Reply:
Thomas: I didn't set sync on my machine so this is not a required change.